src/Client.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Utopia\Fetch;
+
+/**
+ * Client class
+ * @package Utopia\Fetch
+ */
+class Client
+{
+    public const METHOD_GET = 'GET';
+    public const METHOD_POST = 'POST';
+    public const METHOD_PUT = 'PUT';
+    public const METHOD_PATCH = 'PATCH';
+    public const METHOD_DELETE = 'DELETE';
+    public const METHOD_HEAD = 'HEAD';
+    public const METHOD_OPTIONS = 'OPTIONS';
+    public const METHOD_CONNECT = 'CONNECT';
+    public const METHOD_TRACE = 'TRACE';
+    /**
+     * Flatten request body array to PHP multiple format
+     *
+     * @param array<mixed> $data
+     * @param string $prefix
+     * @return array<mixed>
+     */
+    private function flatten(array $data, string $prefix = ''): array
+    {
+        $output = [];
+        foreach ($data as $key => $value) {
+            $finalKey = $prefix ? "{$prefix}[{$key}]" : $key;
+
+            if (is_array($value)) {
+                $output += $this->flatten($value, $finalKey); // @todo: handle name collision here if needed
+            } else {
+                $output[$finalKey] = $value;
+            }
+        }
+
+        return $output;
+    }
+    /**
+     * This method is used to make a request to the server
+     * @param string $url
+     * @param array<string, string> $headers
+     * @param string $method
+     * @param array<string>|array<string, mixed> $body
+     * @param array<string, mixed> $query
+     * @return Response
+     */
+    private function buildResponse(
+        string $url,
+        array $headers,
+        string $method,
+        array $body,
+        array $query
+    ): Response {
+        // Process the data before making the request
+        if(!$method) { // if method is not set, set it to GET by default
+            $method = self::METHOD_GET;
+        } else { // else convert the method to uppercase
+            $method = strtoupper($method);
+        }
+        // if there are no headers but a body set header to json by default
+        if(!isset($headers['content-type']) && is_array($body) && count($body) > 0) {
+            $headers['content-type'] = 'application/json';
+        }
+        if(isset($headers['content-type'])) {
+            match ($headers['content-type']) { // Convert the body to the appropriate format
+                'application/json' => $body = json_encode($body),
+                'application/x-www-form-urlencoded' => $body = $this->flatten($body),
+                'multipart/form-data' => $body = $this->flatten($body),
+                'application/graphql' => $body = $body[0],
+                default => $body = $body,
+            };
+        }
+        if(!is_array($headers)) {
+            $headers = [];
+        }
+        foreach ($headers as $i => $header) { // convert headers to appropriate format
+            $headers[] = $i . ':' . $header;
+            unset($headers[$i]);
+        }
```

Comment:
Instead of this you can use 

```php
$headers = array_map(function ($i, $header) {
  return $i . ':' . $header;
}, array_keys($headers), $headers);
```